Makefile

```diff
@@ -25,36 +31,38 @@ packs-tests: requirements .clone_st2_repo .packs-tests
 	@echo
 	@echo "==================== pylint ===================="
 	@echo
-	. $(VIRTUALENV_DIR)/bin/activate; find ${ROOT_DIR}/packs/* -maxdepth 0 -type d -print0 | xargs -0 -I FILENAME scripts/pylint-pack.sh FILENAME
+	# Scans directory for all changes in current branch from master, and filters out only packs that have changed.
+	. $(VIRTUALENV_DIR)/bin/activate; for pack in "$(CHANGED_PACKS)"; do if [ -n "$$pack" ]; then scripts/pylint-pack.sh $$pack; fi; done
 
 .PHONY: flake8
 flake8: requirements
 	@echo
 	@echo "==================== flake8 ===================="
 	@echo
-	. $(VIRTUALENV_DIR)/bin/activate; find ${ROOT_DIR}/packs/* -name "*.py" -print0 | xargs -0 flake8 --config ./.flake8
+	# Scans directory for all changes in current branch from master, and filters out only python files that have changed.
+	. $(VIRTUALENV_DIR)/bin/activate; for file in "${CHANGED_PY}"; do if [ -n "$$file" ]; then flake8 --config ./.flake8 $$file; fi; done
 
 .PHONY: configs-check
 configs-check: requirements
 	@echo
 	@echo "==================== configs-check ===================="
 	@echo
-	. $(VIRTUALENV_DIR)/bin/activate; find ${ROOT_DIR}/packs/* -name "*.json" -print0 | xargs -0 -I FILENAME ./scripts/validate-json-file.sh FILENAME
-	. $(VIRTUALENV_DIR)/bin/activate; find ${ROOT_DIR}/packs/* -name "*.yaml" -print0 | xargs -0 -I FILENAME ./scripts/validate-yaml-file.sh FILENAME
+	. $(VIRTUALENV_DIR)/bin/activate; for file in "$(CHANGED_YAML)"; do if [ -n "$$file" ]; then ./scripts/validate-yaml-file.sh $$file; fi; done
```

Comment:
I'm cherry picking and modifying those changes and I just noticed there is a bug here.

Since the whole string is quoted if there are multiple changed files it will actually only iterate once over the whole string (and not over all the words).